sdk/communication/Azure.Communication.Calling.Server/src/CallClient.cs

```diff
@@ -0,0 +1,604 @@
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callLegId"/> is null.</exception>
+        public virtual Response DeleteCall(string callLegId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(DeleteCall)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNullOrEmpty(callLegId, nameof(callLegId));
+
+                return RestClient.DeleteCall(callLegId, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary> Disconnect the current caller in a Group-call or end a p2p-call asynchronously.</summary>
+        /// <param name="callLegId"> The call leg id. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callLegId"/> is null.</exception>
+        public virtual async Task<Response> HangupCallAsync(string callLegId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(HangupCall)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNullOrEmpty(callLegId, nameof(callLegId));
+                return await RestClient.HangupCallAsync(callLegId, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary> Disconnect the current caller in a Group-call or end a p2p-call. </summary>
+        /// <param name="callLegId"> The call leg id. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callLegId"/> is null.</exception>
+        public virtual Response HangupCall(string callLegId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(HangupCall)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNullOrEmpty(callLegId, nameof(callLegId));
```

Comment:
According to the .Net guidelines, we should not validate service parameters, only client side parameters: https://azure.github.io/azure-sdk/dotnet_introduction.html#parameter-validation

Reply:
I see this is a common pattern across all the functions, please take a look :)

Reply:
call leg id here is indeed a service parameter according to the definition. 

But I think we should failed fast on the client side for the invalid parameter error. @navali-msft @ajaych-ms , what do you think if it?

Reply:
It's Azure guideline to let the service validate parameters, so we have to remove it here. The reason being that developers use raw REST APIs without the SDK and if we implemented client-side validations, clients easily start diverge across languages and with the service.

Reply:
Removed all service parameters validation.